### PR TITLE
fix(AWS Lambda Layers): Permissions on lambda layer retained

### PR DESCRIPTION
**Pull request by raym0nd93**
Closes: #8578

### CODE REVIEW

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @raym0nd93 ! Does it confirm to fix the issue on your side?

Can we configure one regression test in scope of this describe block: https://github.com/serverless/serverless/blob/9bdcb4fc17ad06faf0c666f33850fee6d3cd6a6c/test/unit/lib/plugins/aws/package/compile/layers/index.test.js#L360

**Comment by raym0nd93** [comment]
Yes I've tested the code as it is in the MR, it did retain the permission on the lambda layer.

**Comment by raym0nd93** [comment]
I'm not 100% sure if this is what you meant, any feedback is very welcome

File: test/unit/lib/plugins/aws/package/compile/layers/index.test.js [hunk 1]
```diff
@@ -366,6 +366,16 @@ describe('lib/plugins/aws/package/compile/layers/index.test.js', () => {
...
+
+      const layerPermissionResourceNamePrefix = naming.getLambdaLayerPermissionLogicalId(
+        'layerRetain'
+      );
+      const layerPermissionResourceName = Object.keys(cfResources).find(resourceName =>
+        resourceName.startsWith(layerPermissionResourceNamePrefix)
+      );
+      expect(layerPermissionResourceName).to.not.equal(layerPermissionResourceNamePrefix);
+      const layerPermissionResource = cfResources[layerPermissionResourceName];
+      expect(layerPermissionResource.DeletionPolicy).to.equal('Retain');
```
Comment (medikoo) [inline test/unit/lib/plugins/aws/package/compile/layers/index.test.js:378]: Let's put it in another `it`, so it serves as individual spec item.

I'd probably name it: _should retain permission resource_

**Review by medikoo — CHANGES_REQUESTED** [review]
@raym0nd93 thanks for update! Please see my comment, also note that CI started to fail.

Did you intentionally converted the PR into draft?

**Comment by pgrzesik** [comment]
Hello @raym0nd93, thanks a lot for your contributions to this PR :bow: We'd like to wrap it up in the coming days, so I hope you don't mind me taking over your work and applying the finishing changes. Please let me know if you'd still like to continue your work here. Once again, your help is much appreciated :bow:

**Comment by pgrzesik** [comment]
Closing this one in favor of https://github.com/serverless/serverless/pull/8688

**Pull request closed without merging**